Makefile

```diff
@@ -1,5 +1,6 @@
 BASE_BRANCH ?= devel
 export BASE_BRANCH
+export VERSION="devel"
```

Comment:
This export is not used by any code. Why did you add it?

Reply:
@cyclinder you did not answer yet.

Any reason to export this variable and have a default value?

Previously there was no default value and the in the code we had a default value
to make it clear that the program was not compiled correctly. The purpose of this
change is to log version info in all submariner components, not to change the way
we log submariner version.

It is best to keep existing behavior if you don't have a good reason to change it,
and if you have a reason to change it should be in a separate commit. Every commit
should do one logical change.

Reply:
Hi @nirs, Sorry for this, I missed it.

But I tried to confirm if my changes actually worked, I'm trying to build submariner on my premises. but it failed. As mentioned here:  https://github.com/submariner-io/submariner/pull/2552#discussion_r1241389145

And I  use `go build` to compile the binary on my premises, I found that there is no build infos in the logs：

```shell
root@cyclinder3:~/submariner# CGO_ENABLED=0 go build  -ldflags "-X github.com/submariner-io/submariner/pkg/version/version.Version=devel -X github.com/submariner-io/submariner/pkg/version/version.GitCommitHash=d18da3d16e23346e16c3998ea7666040232b3f1c -X github.com/submariner-io/submariner/pkg/version/version.GitCommitDate=2023-06-26T10:02:45+08:00 -X github.com/submariner-io/submariner/pkg/version/version.test=test" -o submariner-gateway .
root@cyclinder3:~/submariner#
root@cyclinder3:~/submariner#
root@cyclinder3:~/submariner# ./s
scripts/            submariner-gateway
root@cyclinder3:~/submariner# ./submariner-gateway
2023-07-02T03:44:08.086Z INF ../submariner/main.go:119 main                 Starting the submariner gateway engine
2023/07/02 03:44:08 log===============:
2023-07-02T03:44:08.086Z INF ..g/version/version.go:38 main                 logger===============
2023-07-02T03:44:08.086Z INF ..g/version/version.go:39 main                 Go Version: go1.20.4
2023-07-02T03:44:08.086Z INF ..g/version/version.go:40 main                 Go Arch: amd64
2023-07-02T03:44:08.086Z INF ..g/version/version.go:41 main                 Git Commit Hash:
2023-07-02T03:44:08.086Z INF ..g/version/version.go:42 main                 Git Commit Date:
2023-07-02T03:44:08.086Z INF ..g/version/version.go:43 main                 Submariner Version:
2023-07-02T03:44:08.087Z INF ../submariner/main.go:129 main                 Parsed env variables
W0702 03:44:08.192381 3157994 client_config.go:618] Neither --kubeconfig nor --master was specified.  Using the inClusterConfig.  This might not work.
W0702 03:44:08.192458 3157994 client_config.go:623] error creating inClusterConfig, falling back to default config: unable to load in-cluster configuration, KUBERNETES_SERVICE_HOST and KUBERNETES_SERVICE_PORT must be defined
2023-07-02T03:44:08.192Z FTL ../submariner/main.go:148 main                 Error building kubeconfig error="invalid configuration: no configuration has been provided, try setting KUBERNETES_MASTER environment variable"
```

Reply:
Looks like a good test to verify the changes if building fails.

I think there are few issues in the way you refer to version package:

    -ldflags "-X github.com/submariner-io/submariner/pkg/version/version.Version=devel"

- Using the name of the public function `Version` instead of the private variable `version`
- Using the path to the file `pkg/version/version` instead the of the package name `submariner/version`.

Maybe this works?

    -ldflags "-X github.com/submariner-io/submariner/version.version=devel"

This tutorial seems to cover this case:
https://www.digitalocean.com/community/tutorials/using-ldflags-to-set-version-information-for-go-applications#targeting-sub-package-variables

Reply:
I found the reason, it should be like like:

```
-ldflags "-X github.com/submariner-io/submariner/pkg/version.version=devel"
```

I did a test locally and it did work:

```
root@cyclinder3:~/submariner# CGO_ENABLED=0 go build  -ldflags "-X github.com/submariner-io/submariner/pkg/version.version=devel -X github.com/submariner-io/submariner/pkg/version.gitCommitHash=d18da3d16e23346e16c3998ea7666040232b3f1c -X github.com/submariner-io/submariner/pkg/version.gitCommitDate=2023-06-26T10:02:45+08:00" -o submariner-gateway .
root@cyclinder3:~/submariner# ./submariner-gateway
2023-07-03T15:14:37.805Z INF ../submariner/main.go:119 main                 Starting the submariner gateway engine
2023-07-03T15:14:37.806Z INF ..g/version/version.go:39 main                 Go Version: go1.20.4
2023-07-03T15:14:37.806Z INF ..g/version/version.go:40 main                 Go Arch: amd64
2023-07-03T15:14:37.806Z INF ..g/version/version.go:41 main                 Git Commit Hash: d18da3d16e23346e16c3998ea7666040232b3f1c
2023-07-03T15:14:37.806Z INF ..g/version/version.go:42 main                 Git Commit Date: 2023-06-26T10:02:45+08:00
2023-07-03T15:14:37.806Z INF ..g/version/version.go:43 main                 Submariner Version: devel
2023-07-03T15:14:37.807Z INF ../submariner/main.go:129 main                 Parsed env variables
W0703 15:14:37.808158 3212364 client_config.go:618] Neither --kubeconfig nor --master was specified.  Using the inClusterConfig.  This might not work.
W0703 15:14:37.808224 3212364 client_config.go:623] error creating inClusterConfig, falling back to default config: unable to load in-cluster configuration, KUBERNETES_SERVICE_HOST and KUBERNETES_SERVICE_PORT must be defined
2023-07-03T15:14:37.808Z FTL ../submariner/main.go:148 main                 Error building kubeconfig error="invalid configuration: no configuration has been provided, try setting KUBERNETES_MASTER environment variable"
```